src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -293,22 +304,36 @@ private void EmitTypeDeclaration(ExpressionEmitter emitter, DeclaredTypeExpressi
             emitter.EmitObjectProperty(declaredType.Name,
                 () =>
                 {
-                    EmitProperties(emitter, ApplyTypeModifiers(declaredType, TypePropertiesForTypeExpression(declaredType.Value)));
+                    var declaredTypeObject = ApplyTypeModifiers(declaredType, TypePropertiesForTypeExpression(declaredType.Value));
+                    if (ImportClosureInfo.ImportedSymbolOriginMetadata.TryGetValue(declaredType.Name, out var originMetadata))
+                    {
+                        var importedFromProperties = ExpressionFactory.CreateObjectProperty(LanguageConstants.ImportMetadataSourceTemplatePropertyName,
+                            ExpressionFactory.CreateStringLiteral(originMetadata.SourceTemplateIdentifier)).AsEnumerable();
```

Comment:
I'm a bit concerned that we might be exposing too much information here if the symbol comes from a local file, a template spec, or a private module registry. 

The format used will either be a path relative to the entrypoint model or a fully qualified `ts:` or `br:` URI, and the intention is to help with debugging (so that a user can see that a type named `1.foo` in their compiled template came from `mod.bicep` or `br:mcr.microsoft.com:network/virtual-network:1.0.1`). 

This data is also useful for decompilation when the source template is in `br/public`, but there's always a chance a local file or private registry module could have an embarrassing or too-revealing name, like `templates_from_people_I_am_about_to_fire/mod.bicep` or `br:acr.microsoft.com:companyName/templates_we_stole_from_rival_x/storage:1.0.1`

Reply:
Personally, I think it would fine to emit the source template information. The the same information already exists in the Bicep file, and the generated template might already contain other confidential data, so emitting the source template information does not inherently introduce new risks. We just need to make sure not to accidentally log the property in the backend.